### ISSUE TITLE
1.12 port

### ISSUE THREAD

**Issue by xbony2** [Belgabor/DrawersBits#12]
Kappa

(I might attempt it myself if you are officially disappeared)

**Comment by Belgabor** [comment]
I wanted to do it a while ago only to find C&B hadn't been updated yet XD. I guerss that changed now, so I'll take a look.

**Comment by xbony2** [comment]
cool \o/

**Comment by Draskh** [comment]
any progress?

**Comment by ResqDiver1317** [comment]
I'm wondering this myself. I'm currently putting together a modpack for our community that has a strong focus on building without all the fancy tech mods and was wondering if this is close to being updated to 1.12.2 or not. Many of our players use the bit drawers on our 1.10.2 servers so it would be awesome to have it for our 1.12.2 as well. Thanks!

**Comment by kreezxil** [comment]
Is belgabor still among the living?

**Comment by Belgabor** [comment]
Yeah, sorry, just been busy with life recently XD
I mean to get back to it but haven't be able to far.

**Comment by JelmarG** [comment]
A port would definitely be appreciated. C&B is pretty hard to use without Drawers & Bits.

**Comment by GuntherDW** [comment]
Update requests are kind of looked down upon, but I'll check out the repository and see if getting it up and running for 1.12 is going to be a hassle.
I'll send a PR your way when I think it's of decent enough quality.

**Comment by Belgabor** [comment]
That would be great, my time is currently limited and I'm fairly rusty on modding by now.

**Comment by GuntherDW** [comment]
Work is progressing steadily.
There's a bunch of issues that I'm working on right now, but an initial 1.12 port shouldn't take *too* long.
Storage Drawers did get quite a refactor though, so converting the codebase to support that stuff can be troublesome.

![screenshot](https://i.imgur.com/BlPRn0G.png)

**Comment by GuntherDW** [comment]
We've had the mod running on a server now for about 2 weeks, and haven't really found any issues yet that weren't also in the original storage drawers mod.

The code quality doesn't look as pretty as I'd like it to be, being that I've had to copy&paste a bunch of code from the storage drawers codebase (just like you had to in the beginning), but at least's it's a start.

**Comment by Belgabor** [comment]
Awesome. How do you want to proceed? Do you want to PR it? If you want I can also instate you as a coauthor here and on Curse so you can work directly on the code and release.

**Comment by GuntherDW** [comment]
In its current state it's rather... ugly, but workable.
In order to really clean up the code a more direct co-operation between the Storage Drawer devs should yield some better results.
However this is a good starting point.

I could PR my current code, but like I said to be perfectly honest it really needs some work still.
It could at least open it up for more bugreports and fixes if we allow other players to play with it.

I'm not sure how co-author stuff on Curse works, but that's something for you to decide.

**Comment by Belgabor** [comment]
I agree, let's get it out there. I'm still away from my development PC
until late Thursday though, so no rush

GuntherDW <notifications@github.com> schrieb am So., 1. Juli 2018 06:42:

**Comment by Kirin92** [comment]
It looks like no further progress here?

**Comment by xbony2** [comment]
Nah

**Comment by Belgabor** [comment]
I unfortunately never heard back from them and personally I currently lack the time to do a full port myself.

**Comment by JelmarG** [comment]
@GuntherDW was working on it.
